src/Compilers/CSharp/Portable/Parser/Lexer.cs

```diff
@@ -2422,34 +2421,31 @@ private void LexSyntaxTrivia(bool afterFirstToken, bool isTrailing, ref SyntaxLi
 
         private bool IsConflictMarkerTrivia()
         {
-            var position = TextWindow.Position;
-            var text = TextWindow.Text;
-            if (position == 0 || SyntaxFacts.IsNewLine(text[position - 1]))
-            {
-                var firstCh = text[position];
-                Debug.Assert(firstCh == '<' || firstCh == '=' || firstCh == '>');
+            var firstCh = TextWindow.PeekChar();
+            Debug.Assert(firstCh is '<' or '=' or '>');
 
-                if ((position + s_conflictMarkerLength) <= text.Length)
+            for (int i = 1, n = s_conflictMarkerLength; i < n; i++)
+            {
+                if (TextWindow.PeekChar(i) != firstCh)
                 {
-                    for (int i = 0, n = s_conflictMarkerLength; i < n; i++)
-                    {
-                        if (text[position + i] != firstCh)
-                        {
-                            return false;
-                        }
-                    }
-
-                    if (firstCh == '=')
-                    {
-                        return true;
-                    }
-
-                    return (position + s_conflictMarkerLength) < text.Length &&
-                        text[position + s_conflictMarkerLength] == ' ';
+                    return false;
                 }
             }
 
-            return false;
+            if (firstCh != '=' && TextWindow.PeekChar(s_conflictMarkerLength) != ' ')
+            {
+                return false;
+            }
+
+            // Keep the new line check last because TextWindow.Reset might need to recopy the buffer,
+            // although that's rare in practice.
+            return TextWindow.Position == 0 || SyntaxFacts.IsNewLine(getLastChar());
```

Comment:
Really rare. At first, I kept this check at the top and tried debugging when parsing Roslyn's Syntax.xml.Internal.Generated.cs and a few other large files, and never hit the breakpoint in `TextWindow.Reset` where it had to copy stuff (and the benchmark results were the same). So I could copy the check back to the top if someone feels that is better.